# Methods

## Preparation of reads from fungal isolates for synthetic dataset

In order to validate the pipeline, a mock dataset has been created to simulate scenarios of where multiple fungal species are present in a sample.
Using reads from ONT sequencing of 66 known fungal isolates, we are able to control the proportions of species for each simulated scenario.
These synthetically generated datasets provide insight into the limitations and sensitivity of this pipeline.
Supplemental Table \@ref(tab:samplesheet) provides the list of fungal species/strains and the number of raw reads for each.
The steps to generate and prepare reads for this synthetic dataset are outlined below.

### DNA extraction, amplification and sequencing
The genomic DNA of 66 fungal strains was extracted with the Qiagen DNeasy Plant mini kit.
 from mycelia or spore material.
Partial SSU, full ITS region and partial LSU regions of each sample were amplified with primers NS5 (forward) and LR6 (reverse).
The amplicons were prepared with the Native Barcoding Kit 96 V14 (code SQK-NBD114.96, ONT).
All samples were multiplexed and sequenced using the MinION R10.4 flow cell.

### Bioinformatics - Basecalling, trimming and filtering
The raw ONT data was basecalled and demultiplexed using Guppy v6.4.2 with the super-high accuracy model (dna_r10.4.1_e8.2_400bps_sup, version unknown).
The mean read length of amplicons after basecalling was ~2.2 KB as in Figure \@ref(fig:rawReadQuality).

- [alt] Dorado: dna_r10.4.1_e8.2_400bps_sup@v4.1.0

Adapters were trimmed from raw basecalled reads using Dorado v0.6.1 with the '`--no-trim-primers`' to avoid removing primer sequences.
Cutadapt v4.6 (Martin, 2011) has been used to select and trim primers from amplicons that contain both forward (NS5) and reverse primer (LR6) sequences.
Amplicons that do not contain both of these primer sequences were excluded from the analysis.
Cutadapt was also used to re-orient reads that have been sequenced by the reverse strand ('`--revcomp`' option), making it easier to process by downstream tools.

The full ITS region of these reads was extracted using ITSxpress v2.0.1 with default settings other than '`--single_end`' and '`--taxa Fungi`' options.
Chopper v0.7.0 was used to select reads of the full ITS region having a length between 300-6000bp and mean Phred quality score above Q20.
Chimeric reads were detected using VSEARCH's denovo and reference based methods.
The database used for reference based chimera detection of full ITS sequence was the UNITE general release v9.0 (sh_general_release_dynamic_s_all_25.07.2023).

### Synthetic dataset - Scenario 1 - Even sampling

- Even distribution
  - 60 samples - subsampled to meet library sizes: 1200, 3000, 10020, 60000, 120000, 150000. 20, 50, 167, 1000, 2000, 2500 reads per sample.
  - 59 species (two Austropuccinia strains)
- subsampling
  - seqkit v1.4
  - 5 repetitions, deterministic seed values for reproducibility

### Synthetic dataset - Scenario 2 - Uneven sampling (5 low abundance)

- 5 samples with very low, rest high

### Synthetic dataset - Scenario 3 - Uneven sampling (5 high abundance)

- all very low, 5 high

### Synthetic dataset - Scenario 4 - Linear range

## Bioinformatics - Clustering

- vsearch at 97% identity
  - OTU filtering as proportion of library size (0.1%). See Results for determination
  - ?Sample-wise OTU filtering - index bleed?
- nanoclust
  - kmer size 6
  - umap parameters (epsilon 0.5)
  - minimum cluster size - See results for determination Section \@ref(cluster-results)
  - selection of most abundant sequence in each cluster

## Taxonomic assignment
