# Methods

## Preparation of reads for synthetic dataset

intro context/goal Table \@ref(tab:samplesheet)

### Sequencing
- Full ITS sequences amplified from mycelia or spores of 66 fungal strains
- PCR primers: ITS_NS5(F)/ITS_LR6(R)
- Nanopore MinION R10.4 chemistry
- Kit SQK-NBD114.96 - multiplexed

### Bioinformatics - Trimming and filtering
- Guppy 6.4.2
  - super-high accuracy dna_r10.4.1_e8.2_400bps_sup
- [alt] Dorado: dna_r10.4.1_e8.2_400bps_sup@v4.1.0
- adapter trimming
- primer trimming
- barcode region extraction
  - full ITS
- quality and length filtering
- chimera filtering
  - denovo, reference based (UNITE)

### Synthetic dataset - Scenario 1 - Even sampling

- Even distribution
  - 60 samples - subsampled to meet library sizes: 1200, 3000, 10020, 60000, 120000, 150000. 20, 50, 167, 1000, 2000, 2500 reads per sample.
  - 59 species (two Austropuccinia strains)
- subsampling
  - seqkit
  - 5 repetitions, deterministic seed values for reproducibility

### Synthetic dataset - Scenario 2 - Uneven sampling (5 low abundance)

- 5 samples with very low, rest high

### Synthetic dataset - Scenario 3 - Uneven sampling (5 high abundance)

- all very low, 5 high

### Synthetic dataset - Scenario 4 - Linear range

## Bioinformatics - Clustering

- vsearch at 97% identity
  - OTU filtering as proportion of library size (0.1%). See Results for determination
  - ?Sample-wise OTU filtering - index bleed?
- nanoclust
  - kmer size 6
  - umap parameters (epsilon 0.5)
  - minimum cluster size - See results for determination Section \@ref(cluster-results)
  - selection of most abundant sequence in each cluster

## Taxonomic assignment
