---
title: "Results"
---

## Quality control

After preparing the reads from fungal isolates for mock scenarios there was considerable read loss across many of the samples (@fig-readCountDistributionBySample).
Seven samples were excluded when creating mock scenarios due to having a low number of reads relative to other samples.
Sample with greater than 2500 reads were kept for the mock scenarios so that we could test the pipeline with a library size of around 150K reads.

::: {#fig-readCountDistributionBySample}
```{r}
knitr::include_graphics('analysis/images/06-read-count-distribution-by-sample.png')
```
:::{custom-style="Caption"}
The distribution of read counts per sample before (A) and after (B) quality control processing. Reads having less than 2500 reads were excluded from mock scenarios (dotted line in B). Note the x-axis is on a log scale.
:::

Read count loss by sample after quality control
:::

Tracking read counts at each stage of the pipeline has shown that the majority of reads were lost in the primer trimming and quality filtering stages of the pipeline (@fig-readLossByStage).
An average of 32.37% of reads were lost across samples after applying cutadapt to select and trim amplicons that contain both forward (ITS1F) and reverse primer (LR3) sequences.
The extraction of full ITS regions led to an average loss of 10.65% trimmed reads across samples.
An average of 71.82% of full ITS sequences were lost after selecting reads between 300-6000bp in length and having a mean quality above Q20 (Phred scale).
Such large loss of reads can be attributed to many of the reads from the dataset being below the minimum mean quality threshold of Q20 as in @fig-rawReadQuality.

::: {#fig-readLossByStage}
```{r}
knitr::include_graphics('analysis/images/06-read-loss-by-stage-by-sample.png')
```
:::{custom-style="Caption"}
Read loss at each filtering and trimming stage of the pipeline. Each point represents a fungal isolate sample and samples that were excluded from mock scenarios due to low read count have been coloured red. The blue horizontal bar with percantages indicate the mean proportion of reads lost from the previous step across the included samples. Each step was executed in order from left to right.
:::

Read loss by pipeline stage
:::

:::{#fig-rawReadQuality}

![](analysis/images/06-raw-read-quality.png)

:::{custom-style="Caption"}
The mean read quality of unprocessed reads before trimming or filtering steps. Quality scores are in the Phred (Q) scale. Read lengths are shown in thousands of basepairs. Histograms on X and Y axes indicate the density of reads at respective quality scores and read lengths. (Plot generated by NanoPlot [@DeCoster2023])
:::

Read quality of raw reads

:::

## Minimum cluster size threshold can recover expected number of species from even abundance scenario {#sec-cluster-results}

The clustering approach that was adopted aimed to group full ITS sequences from the same species together.
The total number of clusters was used as a measure for successfully estimating number of species in the community.
In this case we considered the even abundance mock community where all isolates were equally represented (Scenario 1).

Clustering using VSEARCH at 97% sequence identity consistently over-estimated the number of species in mixed read scenarios.
Large numbers of OTUs were encountered with many of them having low abundance.
We explored removing these low-abundance OTUs by selecting a minimum OTU size cutoff (@fig-vsearchotus and @fig-compareVsearchNanoclust).
OTUs that had read count which was less than the cutoff were discarded and
the cutoff was specified as a proportion of the total library size.
The minimum OTU size cutoffs were applied across a range of values between 0% and 1% of the total library size.
When clustering with VSEARCH at 97% sequence identity, the threshold that gave consistent recovery of the number of species was 0.15% of total library size.
This threshold was robust across different library sizes (1160 to 145000 total reads) generated in this mock scenario.


:::{#fig-compareVsearchNanoclust}
![](analysis/images/06-otu-count-and-loss.png)

:::{custom-style="Caption"}
Comparing the impact of a minimum cluster size threshold on the number of clusters and read loss between NanoCLUST and VSEARCH clustering methods. Plots have been organised in columns by increasing library size. The top row of plots show the number of clusters after applying a minimum cluster size cutoff that is a proportion of the total library size. The bottow row of plots show the proportion of reads that are lost after applying the minimum cluster size threshold. The mean values of each have been plotted after five random resamplings at each library size. The dashed line indicates the actual number of species in the mock scenario.
:::

Minimum Cluster Size Thresholds
:::

:::{#fig-vsearchotus}
![](analysis/images/06-otu-count-vsearch.png)

:::{custom-style="Caption"}
The effect of a minimum OTU size threshold on the number of OTUs when clustering with VSEARCH. Colours indicate the minimum OTU size threshold as a proportion of the total library size. The dashed line indicates the actual number of species in the mock scenario. Five repetitions were performed for each library size. X and Y axes are on a logarithmic scale.
:::

Minimum Cluster Size - VSEARCH
:::


A k-mer based clustering method (used by NanoCLUST) was explored to test whether we could improve the accuracy of species level clustering and retaining more reads compared to VSEARCH.
The minimum cluster size parameter used by the HDBSCAN significantly impacts the resulting number of clusters by limiting the groupings that can be considered a cluster.
We tested multiple library sizes to determine an optimal value for this minimum cluster size parameter.
When the minimum cluster size parameter was set to its minimum value (2), the NanoCLUST method overestimated the number of species for library sizes ~1000 and above.
The number of clusters increased significantly with library size for small values of the minimum cluster parameter (@fig-nanoclustOTUs).
For library sizes of ~1000 reads and above, a minimum cluster size of 0.65% (of the total library size) approximately recovered the expected number of species.

:::{#fig-nanoclustOTUs}

![](analysis/images/06-otu-count-nanoclust.png)

:::{custom-style="Caption"}
The effect of minimum cluster size parameter and total library size on the number of clusters when following the UMAP and HDBSCAN (NanoCLUST) clustering approach. Colours indicate the minimum cluster size parameter as a proportion of the total library size. The dashed line indicates the actual number of species in the mock dataset. Five repetitions were performed for each library size. X and Y axes are on a logarithmic scale.
:::

Minimum cluster size - NanoCLUST (UMAP + HDBSCAN)
:::

A comparison between the clustering methods was performed by comparing the read loss at a given minimum cluster size cutoff.
Read loss in this instance refers to the proportion of reads that were removed from the analysis due to being placed in a cluster that was smaller than the given threshold.
We found that for low library sizes (20-167 reads per sample), the NanoCLUST method produced closer estimates of the actual number of species
with lower read loss than VSEARCH for all threshold values (@fig-compareVsearchNanoclust).
For larger library sizes (1000-2500 reads per sample), the NanoCLUST method's minimum cluster size of 0.65% produced consistently
accurate estimates of the actual number of species while VSEARCH's minimum cluster threshold of 0.15% performed similarly.
When considering read loss at these thresholds, the NanoCLUST method performed better at 1000 reads per sample, similarly at 2000 reads per sample
and worse at 2500 reads per sample compared to VSEARCH.

### Comparing cluster delineation between VSEARCH and NanoCLUST

To investigate the effect on clustering of closely related species we looked at a single execution of the even abundance scenario (Scenario 1) where 2000 reads were selected from each isolate giving a total library size of 116K reads.
Reads were clustered using the UMAP and HDBSCAN method (NanoCLUST) using a minimum cluster size of 580 (0.5% of the library size).
The most abundant sequence of each cluster was given a taxonomic assignment with dnabarcoder and the UNITE 2024 reference database.

The samples from the order *Pucciniales* are shown in @fig-splittingPuccinia.
We can see that the majority of reads from the *Puccinia striiformis (var tritici)* sample (BC 25) have clustered together into a single cluster (OTU 43),
which has been given the expected species-level classification (*Puccinia striiformis*).
The remaining reads from the *P. striiformis (var tritici)* sample have clustered into two groups (5 reads each) which do not correspond to the expected sample taxonomy.
The taxonomic classification given to these clusters corresponds to other samples present in the library (*Zymoseptoria tritici* and the *Trichomonascus* genus)
and may be indicative of index-switching (where a sequencing error has occurred in the barcode region of the read).

:::{#fig-splittingPuccinia}
![](analysis/images/06-cluster-splitting-nanoclust-puccinia.png)

![](analysis/images/06-cluster-splitting-vsearch-puccinia.png)

:::{custom-style="Caption"}
Two plots showing how reads from Pucciniales isolates are split into clusters using the NanoCLUST (UMAP + HDBSCAN) and VSEARCH methods. Bars indicate the abundance of a cluster (number of reads). Taxonomic classification of each cluster is shown in the X axis labels. Green indicates that a cluster matches the expected species-level classification.
:::

Clustering of Pucciniales - NanoCLUST vs VSEARCH
:::

In the *Puccinia graminis* sample (BC 27), the majority of the reads have been split into two clusters, both of which
have been classified as the expected *P. graminis* species.
This likely indicates biological variation in the *P. graminis* sample that is causing the single sample to be split into separate clusters.
The majority of reads for both *Austropuccinia psidii* samples (BC 28 and 36) have been split into the same two clusters (OTU 38 and 21)
which have been given the classification *Puccinia psidii*.
A similar plot for all samples in this scenario can be seen in @fig-nanoclustSplitting2

- difficulty of clustering fungal ITS regions at species level due to variation
- classification of Austropuccinia psidii -> Puccinia psidii
  - likely an old name in the reference database, meaning the dnabarcoder's classification is correct

- show results for:
  - Candida
  - Cryptococcus
  - Botrytis


- limitation of using single representative sequence

## Using most abundant sequence for taxonomic assignments less variable than using consensus sequence

In order to explore the accuracy of the taxonomic classifications in the pipeline, two metrics were computed for the even abundance scenario (Scenario 1).
The genera classification proportion metric has been defined as $\frac{\text{\# of reads classified at the genera level}}{\text{total \# of reads from isolate}}$.
The genera precision metric has been defined as $\frac{\text{\# of reads classified correctly at the genera level}}{\text{\# of reads classified at the genera level}}$.
These metrics were calculated for each library size and compared by clustering methods in @fig-nanoclustPrecisionAbundance.

:::{#fig-nanoclustPrecisionAbundance}

![](analysis/images/06-precision-nanoclust-abundance.png)

:::{custom-style="Caption"}

:::

Most abundant sequence less variable in precision than consensus - NanoCLUST
:::

- Figure analysis:
  - more variability with consensus sequences
  - higher precision at lower classification proportion for lowest minimum cluster size (2)
  - minimum cluster size introduces its own variability

- limitations of reference database labels, difficult to ensure taxonomic assignments are correct in an automated fashion.

## Sensitivity of cluster thresholds in uneven abundance scenarios (Scenario 2)

- points at which we lose samples
- tradeoff -> inflated number of clusters / noisy
- min cluster size as proportion of library size not robust for uneven sampling approaches

:::{#fig-uneven-min-cluster-thresh-nanoclust}

![](analysis/images/06-uneven-min-cluster-nanoclust.png)

:::{custom-style="Caption"}
The impact of increasing the minimum cluster size parameter for NanoCLUST (UMAP + HDBSCAN) from left to right (values between 2 and 100).
Only clusters containing reads from the five 'low-abundance' fungal isolates are shown.
OTUs and their taxonomic classification (given by dnabarcoder) are shown on the x-axis.
As minimum cluster size increases, reads from 'low-abundance' fungal isolates are merged with those from
unrelated taxa (i.e. reads from *Eutypa lata* are clustered together with those identified as *Puccinia psidii* at minimum cluster size equal to 20).

:::

Impact of min_cluster_size on low abundance clusters in NanoCLUST
:::
