---
title: "Discussion"
---

```{r}
#| message: false
library(dplyr)
library(flextable)

set_flextable_defaults(
        font.size = 10,
        theme_fun = theme_vanilla
)
```

Our results indicate that our workflow is effective for detecting the major fungal taxa from our mock communities
with less clear results in Australian soil samples due to the high rate of unclassified OTUs.
We observed consistent patterns in clustering with VSEARCH and our customised NanoCLUST implementation.
We have tested the impact of a minimum cluster size threshold and found that the lowest cluster size threshold gives the highest classification accuracy at the cost of overestimating species diversity.
We found that an OTU's most abundant sequence was most reliable for classifying OTUs at the genus level.

## Quality control

As expected from ONT sequencing, the reads used to construct our mock scenarios had
considerably higher error rates than those produced by other sequencing platforms such as Illumina and PacBio [@anslanGreatDifferencesPerformance2018a; @tedersooGlobalSoilMycobiome2021; @tedersooPacBioMetabarcodingFungi2018].
With a mean read quality of Q13.2, our approach to quality control aimed to significantly decrease noise by setting stringent quality thresholds.
Setting a minimum quality score of Q20 and requiring both forward and reverse primers to be present on reads resulted in the removal of 85% of the dataset (@fig-readLossByStage).
This QC approach may be overly stringent and bias against some taxa if applied generally in real world scenarios.
Therefore, we aim to make these quality filtering settings configurable in the pipeline so end-users can determine their own acceptable level of error.
It should be noted that the sequences for the mock dataset were generated in 2022 and were not compatible with dorado's latest basecalling model (v5.0.0 at the time of writing).
We observed that using modern ONT chemistry and the latest basecalling model for our soil samples resulted in many more sequences being retained (40% loss versus the 85% loss in the mock scenarios).

- basecalling and chemistry, trend in improving quality, approaching that of illumina

## Clustering

We have demonstrated that VSEARCH and NanoCLUST clustering methods produced comparable results and
both were able to successfully delineate major taxa in our mock scenarios.
NanoCLUST was generally more sensitive to biological variation than VSEARCH in some fungal groups as it split single taxa into multiple OTUs
(for example: *Pucciniales* (@fig-splittingPuccinia), *Cryptococcus* (@fig-splittingCryptococcus) and *Sclerotiniaceae* (@fig-splittingBotrytis)).
The splitting of taxa in NanoCLUST may suggest the presence of ITS haplotype variants in these fungal groups.
This interpretation should be treated with caution as the UMAP dimension reduction technique cannot guarantee the preservation of all structure in the lower dimensional space [@mittalDimensionalityReductionUsing2024; @asnicarMachineLearningMicrobiologists2024].
Closer analysis of the sequences found in each OTU may give further insight into the true source of this variation.
We anticipate that by increasing the clustering identity used by VSEARCH above 97%, these taxa would also be split into multiple OTUs, although this was not tested.

The metrics we used to measure species recovery (number of OTUs, number of species and percentage of reads lost) gave insight into broad patterns of clustering, but had limitations when comparing clustering methods.
We found that the number of OTUs was a good initial measure for comparing clustering methods as it was relatively efficient to compute.
However, after performing taxonomic assignments, we found that using the number of OTUs to measure species diversity tended
to overestimate the number of species present because many OTUs shared the same species-level classifications (@fig-fixed-min-cluster-size-even).
<!-- both methods recovered core species for all library sizes -->

When comparing the effect of a minimum OTU size on our clustering methods,
we observed predictable trends in the number of OTUs that VSEARCH produced for all library sizes (@fig-compareVsearchNanoclust).
Our NanoCLUST implementation showed less stability than VSEARCH in the number of OTUs it produced at high library sizes.
<!-- We suspect that as HDBSCAN's min cluster size parameter too high (%1 of 145000 -->
Care should be taken when interpreting this comparison as the minimum OTU size filtering is not applied in the same way between clustering methods.
For VSEARCH, the minimum OTU size cutoff is applied after clustering on a fixed set of OTUs.
In our NanoCLUST implementation, HDBSCAN recomputed the set of OTUs for each `min_cluster_size` value and is therefore more dynamic than VSEARCH.

We found that interpreting the percentage of reads lost as a measure of the success was misleading.
NanoCLUST often lost fewer reads than VSEARCH but clustered unrelated taxa together as the minimum OTU size cutoff increased (@fig-compareVsearchNanoclust and @fig-uneven-min-cluster-thresh).
The HDBSCAN algorithm had the unexpected behaviour of merging small clusters together as the `min_cluster_size` parameter increased [@lelandmcinnesParameterSelectionHDBSCAN2016].
This demonstrates that care should be taken when optimising parameters.
While low read loss is ideal, it may have unintended implications in the resulting clustering.
<!-- Care should be taken when setting HDBSCAN parameters to reflect the biological implications. -->
The original NanoCLUST pipeline [@rodriguez-perezNanoCLUSTSpecieslevelAnalysis2021] was designed for more conserved 16S regions in Bacteria, parameters validated against 16S data may not be suitable for variable ITS regions in fungi.

Our NanoCLUST implementation was unsuitable for clustering large datasets due to computational limitations.
In our case, attempting cluster 1.8 million reads from our soil dataset with UMAP and HDBSCAN resulted much higher memory use than was possible with our system (using up to 128GB RAM).
By default, the NanoCLUST pipeline [@rodriguez-perezNanoCLUSTSpecieslevelAnalysis2021] on which we modelled our implementation, only selects 100K reads from the given dataset and is expected to use between 32GB-36GB of RAM [@iterGenomicsITERNanoCLUST2024].
Despite these limitations, NanoCLUST is a complex and dynamic clustering approach that shows promise for clustering highly variable ITS sequences.
While VSEARCH uses a fixed similarity metric which is much easier to understand, NanoCLUST may be suited to cluster fungal groups with differing rates of variability at the ITS locus.
<!--
There are limitations of using a single sequence identity threshold to delineate taxa across the entire fungal tree of life [@vuDNABarcodingAnalysis2016; @vuLargescaleGenerationAnalysis2019; @vuDnabarcoderOpensourceSoftware2022].
While dnabarcoder classifies single sequences using different similarity cutoffs by lineage, the clustering approach taken by NanoCLUST may have potential to handle dynamic clustering of fungal lineages
Dnabarcoder is a tool that tackles this problem by predicting multiple sequence identity cutoffs for different fungal clades based on a reference dataset.
Further work - clade level delineation can be modeled using k-mer frequency, retained in dimension reduction and density based clustering.
-->

<!--
- basecall quality -> eventually approaching the use of ASVs (dada2)
- clusters not dataset-dependent, at cost of overestimating abundance. [@Callahan2017; Tedersoo]
TODO citation -->

## Taxonomic assignments

In our mock communities, in the best cases we observed a genera-level classification rate of ~95% and genera-level precision of 82% to 85%.
This demonstrated that in a controlled scenario, a significant proportion of the major taxa were correctly identified.
However, there were a number of major taxa that were consistently misidentified regardless of clustering approach ([@tbl-top-misidentified]).

:::{#tbl-top-misidentified}
```{r}
flextable(
  read.csv('analysis/tables/top-misidentified-nc.csv') %>%
    rename_all(\(x) gsub('_', ' ', x)) %>%
    mutate(
      score = scales::percent(score, 0.01),
      cutoff = scales::percent(cutoff, 0.01)
    )
) %>%
  width(j=1, width=1, unit = "cm") %>%
  width(j=2, width=2.5, unit = "cm") %>%
  width(j=3, width=4, unit = "cm") %>%
  width(j=4, width=1.5, unit = "cm") %>%
  width(j=5, width=4, unit = "cm") %>%
  width(j=6, width=1.5, unit = "cm") %>%
  width(j=7, width=1.5, unit = "cm")
```

Top 10 misidentified OTUs in an even-abundance scenario (2000 reads per taxa) using NanoCLUST.
These misidentifications were also common across VSEARCH trials.
The columns show: the OTU which the taxon was clustered, the barcode identifying the taxon at time of sequencing,
the taxon's 'known' species-level identification, the number of reads from the taxon in the OTU, the classification given by dnabarcoder,
the BLAST identity score against the reference sequence, and the similarity cutoff used by dnabarcoder to classify at the species level.
:::

While we cannot be certain of the cause of these misclassifications, we hypothesise that some may be explained by cross-sample contamination and mislabelling.
Some misclassified reads had high matching BLAST scores for other taxa that were present in the mock community which may indicate that contamination has occurred.
For example: reads from *Kluyveromyces lactis*, *Cryptococcus neoformans* and *Cryptococcus gattii* had high scoring classifications for *Candida albicans*, *Kluyveromyces marxianus*, *Candida orthopsilosis* respectively.
There was also a potential mislabelling of samples where *Diaporthe sp CCL067* and *Asteroma sp CCL060* were swapped.
There were two *Asteroma* species in the mock community, and we expected them to cluster together.
Instead, *Asteroma sp CCL068* and *Diaporthe sp CCL067* were consistently placed in the same OTU.
In addition, the classification for *Asteroma sp CCL060* was in the *Diaporthe* genus with a high (99.59%) identity.
Despite these observations, the mislabelling is unclear because *Asteroma* and *Diaporthe* are both in *Diaporthales* order, so we expect considerable similarity between ITS sequences.

The inconsistency of taxonomic nomenclature was significant limitation of assessing correct taxonomic assignments in an automated way.
The taxa examined in this study had many taxonomic synonyms, complicating the validation process.
A simple approach that only considers the taxonomic labels as-is can lead to false negatives.
Our approach involved the manual (and non-exhaustive) validation of nomenclature for mock community taxa which would not scale well for larger mock communities.

When assessing the precision of taxonomic classifications from our scenarios,
surprisingly, we found that the highest rates of classification with the most correct classifications occurred with minimal OTU filtering for both NanoCLUST and VSEARCH (@fig-precision).
This went against our expectations that low-abundance OTUs were mostly erroneous and would reduce classification accuracy.
We expect that these small OTUs were the result of sequencing errors but still contain sequences with a high level of similarity to the 'correct' reference database entry.
Further investigation for patterns in these low-abundance OTUs may be fruitful for separating signal from noise.

When comparing the classification accuracy of most-abundant sequences versus consensus sequences, we found that
most-abundant sequence gave less variable results (@fig-precision).
We expected the consensus sequence approach used by NanoCLUST to significantly improve classification precision by correcting sequencing errors.
However, we expect that there was considerable biological variation present within OTUs which introduced noise into the resulting consensus sequence.
This issue was exacerbated when computing consensus sequences for OTUs clustered by VSEARCH at 97% identity as only up to 35% of reads were given classifications.
As NanoCLUST was originally intended for more conserved 16S sequences, the consensus approach may need significant adjustment for highly variable ITS sequences.
If we were to refine the consensus sequences construction,
it would be important to determine whether source of errors in the resulting sequence were due to the consensus tool or variation in the OTUs.

<!--
- limitation of VSEARCH with dnabarcoder
  - mismatch with dnabarcoder cutoffs and pipeline VSEARCH 97% cutoff
  - potential to clump unrelated species in same OTU
-->

The real-world soil case study displayed considerably different groups of fungi than those present in our mock scenarios.
The library size of the soil data (1.8 million reads) was much larger than any of the mock scenarios we tested.
and also native australian fungi that are not well represented in reference databases (CITE).
- many OTUs unclassified in soil data
  - dnabarcoder too stringent?
    - aim to have precise classification, over high-level classification
  - limitation of reference database when working with Australian fungi
  - still able to start analysis of fungi present

## Recommendations

- best practices for running
  - minimum sample sizes for recovering rare species
    - we were able to correctly identify a low-abundance taxon of 5 reads to family level, compared to 50 other taxa at 2000 reads (1:400). total library size of around 100K reads. (with nanoclust min cluster size = 2)
    - (could identify taxa with 10 reads to genus level.)
    - VSEARCH had split this taxa into multiple OTUs (size ~2)
    - NanoCLUST only assigned at family level due to representative sequence, could be resolved by assigning more sequences
    - drawback
      - overestimated the number of species 100 (182 OTUs) vs 55
      - high number of unclassified OTUs
    - potential use case of identifying rare pathogens

## Future work
  - taking advantage of long amplicon (LSU, SSU)
    - aid unclassified species
    - phylogenetic methods
  - try other classification methods