---
title: "Discussion"
---

```{r}
#| message: false
library(dplyr)
library(flextable)

set_flextable_defaults(
        font.size = 10,
        theme_fun = theme_vanilla
)
```

Our results indicate that this ONT workflow is effective for detecting the major fungal taxa from our mock communities
with less clear results in Australian soil samples due to a high rate of unclassified OTUs.
We observed consistent patterns of clustering for VSEARCH and NanoCLUST and have tested the impact of a minimum cluster size threshold.
We have found that using the lowest cluster size thresheld gives the highest classification accuracy at the cost of overestimating diversity.
We found that an OTU's most abundant sequence was most reliable for classifying OTUs at the genus level.

## Quality control

Our approach to quality control aimed to considerably reduce noise from error-prone ONT sequences.
In our mock dataset, by setting a minimum quality score of Q20 and requiring both forward and reverse primers to be present on a read resulted in the removal of 85% of the dataset (@fig-readLossByStage).
This QC approach is stringent and could bias against some taxa in a real world scenario, and we aim to make this configurable in pipeline for end users.
The reads for the mock dataset were generated in 2022 are not able to use the latest dorado's basecalling model (v5.0.0 at the time of writing).
We observed that this is less of a problem with modern ONT chemistry and basecalling because in our real world case study we still retained many more sequences (~40% loss of reads).


## Clustering

We have demonstrated that VSEARCH and NanoCLUST clustering methods produced comparable results and
both were able to successfully delineate major taxa in our mock scenarios.
NanoCLUST was generally more sensitive to biological variation than VSEARCH as it split a single taxa into multiple OTUs
(for example: *Pucciniales* in @fig-splittingPuccinia and *Sclerotinia* from the *Botrytis* group).
This is potentially indicative of ITS haplotype variation and could be confirmed by closer analysis of the sequences within each OTU.
We expect that this variation could also be detected with VSEARCH by increasing the percent identity threshold.

Our metrics of species recovery (number of OTUs, number of species and percentage of reads lost) gave insight into broad patterns of clustering,
but had limitations when comparing VSEARCH and NanoCLUST clustering methods.
We found that the percentage of reads lost as a measure of the success was misleading because
NanoCLUST often produced lower read loss than VSEARCH but in some cases clustered unrelated taxa together (@fig-compareVsearchNanoclust and @fig-uneven-min-cluster-thresh).
This suboptimal clustering of unrelated taxa can be explained by HDBSCAN clustering algorithm:
As the minimum cluster size parameter is increased, nearby clusters are merged together.
<!-- When taken too far, clustering will deviate from biological reality as it using an indirect mapping of kmer profile -> UMAP -->
<!-- Therefore, when set too high, the minimum cluster size parameter -->
Due to dimension reduction via UMAP, it is not necessarily easy to interpret distance.
Care should be taken when setting this and other HDBSCAN parameters to reflect the biological implications.
NanoCLUST originally designed for more conserved 16S regions in Bacteria, parameters validated against 16S data may not be suitable for variable ITS regions in fungi.

We found that the number of OTUs was a good initial measure

- over-estimated species diversity, as confirmed after performing taxonomic assignments + collapsing OTUs @fig-fixed-min-cluster-size-even.
- VSEARCH performed consistently across all library sizes
- NanoCLUST became more variable at high library sizes, suspect due to setting min cluster size too high @fig-compareVsearchNanoclust. (unstable)
- when looking at number of species, both methods recovered core species for all library sizes


NanoCLUST was unsuitable for clustering large datasets due to computational limitations.
In our case, the soil sample dataset contained 1.8 million reads resulted much higher memory use than was possible with our attempts (maximum 128GB).
By default, the NanoCLUST pipeline only selects 100K reads from the given dataset and is expected to use between 32GB-36GB of RAM [@iterGenomicsITERNanoCLUST2024].
<!-- - The NanoCLUST pipeline defaults to subsampling 100K reads from the given dataset.
  - Depending on use case, may not have enough depth -->

Despite some limitations, NanoCLUST is a complex and dynamic clustering approach that shows promise for clustering highly variable ITS sequences.
While VSEARCH uses a fixed similarity metric which is much easier to understand, NanoCLUST may be better suited to cluster fungal groups that contain different rates of variability at the ITS locus.
There exist limitations of using a single sequence identity threshold to delineate taxa across the entire fungal tree of life [@vuDNABarcodingAnalysis2016; @vuLargescaleGenerationAnalysis2019; @vuDnabarcoderOpensourceSoftware2022].
<!--
While dnabarcoder classifies single sequences using different similarity cutoffs by lineage, the clustering approach taken by NanoCLUST may have potential to handle dynamic clustering of fungal lineages
-->
Dnabarcoder is a tool that tackles this problem by predicting multiple sequence identity cutoffs for different fungal clades based on a reference dataset.
Further work - clade level delineation can be modeled using k-mer frequency, retained in dimension reduction and density based clustering.

- basecall quality -> eventually approaching the use of ASVs (dada2)
- clusters not dataset-dependent, at cost of overestimating abundance. [@Callahan2017; Tedersoo]
<!-- TODO citation -->

## Taxonomic assignments

In our mock communities, in the best cases we observed a classification rate of ~95% and genera-level precision between 82-85%.
This demonstrated that in a controlled scenario a significant proportion of the major taxa were correctly identified.
However, there were a number of major taxa that were consistently misidentified regardless of clustering approach ([@tbl-top-misidentified]).

:::{#tbl-top-misidentified}
```{r}
flextable(
  read.csv('analysis/tables/top-misidentified-nc.csv') %>%
    rename_all(\(x) gsub('_', ' ', x)) %>%
    mutate(
      score = scales::percent(score, 0.01),
      cutoff = scales::percent(cutoff, 0.01)
    )
)
```

Top 10 misidentified OTUs in even-abundance scenario (2000 reads per taxa) using NanoCLUST.
These misidentifications were also common across VSEARCH trials as well.
The columns show: the OTU which the taxon was clustered, the barcode identifying the taxon at time of sequencing,
the taxon's 'known' species-level identification, the number of reads from the taxon in the OTU, the classification given by dnabarcoder,
the BLAST identity score against the reference sequence, and the similarity cutoff used by dnabarcoder to classify at the species level.
:::

- potential contamination for some taxa:
  - Kluyveromyces marxianus, Candida orthopsilosis and Candida albicans are taxa that are present in the mock community from other samples.
- potential mislabelling of samples:  Diaporthe sp and Asteroma sp #2
  - expect both Asteroma sp. to cluster together at 97%
  - classification for Asteroma sp. #2 is a Diaporthe genus
    - although Asteroma and Diaporthe both in Diaporthales order, so expect them to be somewhat similar
- Difficult to be conclusive, finer grain look at sequences in the sample may give further insight.

A significant limitation of assessing correct taxonomic assignments in an automated way is the inconsistency of taxonomic nomenclature.
Fungal taxa often have many taxonomic synonyms which makes the process of validating the output of a classifier error-prone.
A simple approach that only considers the taxonomic labels as-is can lead to false negatives as it does not take into consideration - a name's history, or changes.

Our approach involved the manual (and non-exhaustive) validation of nomenclature for mock community taxa which would not scale well for larger mock communities.

In terms of the precision of taxonomic classifications from our scenarios,
surprisingly, we found the lowest rate of unclassified taxa and highest rate of correct taxonomic assignments occurred at minimal OTU filtering for NanoCLUST and VSEARCH (@fig-precision).
This went against our expectations that low-abundance OTUs were mostly erroneous and would produce false classifications.
We expect that these small OTUs are the product of sequencing errors that make them different from major OTUs but still have sequences that are similar to the 'correct' reference database entry.
Further investigation for patterns in these low-abundance OTUs may be fruitful for separating signal from noisy spurious OTUs.

When comparing the classification accuracy of most-abundant sequences versus consensus sequences (@fig-precision), we found that
most-abundant sequence gave less variable results.
We expected the consensus sequence approach used by NanoCLUST to drastically improve classification precision by correcting sequencing errors.
However, due to increased variability, we expect that there was significant biological variation present within OTUs which in turn introduced noise from the consensus algorithm.

  - poor performance for vsearch

As NanoCLUST was originally intended for more conserved 16S sequences, the consensus approach may need significant adjustment for highly variable ITS sequences.
If we were to pursue consensus sequences in the future, it would be important to determine source of errors in the resulting sequence.
Are racon or medaka ideal for this approach?
Also, we would closer look at OTU composition at the sequence level to determine if OTUs are suitable for consensus error correction (do they align).


The real-world soil case studies produces vastly different outcomes than our mock scenarios.
They had much larger library size (3million) and also native australian fungi that are not well represented in reference databases (CITE).
- many OTUs unclassified in soil data
  - dnabarcoder too stringent?
    - aim to have precise classification, over high-level classification
  - limitation of reference database when working with Australian fungi

- limitation of VSEARCH with dnabarcoder
  - mismatch with dnabarcoder cutoffs and pipeline VSEARCH 97% cutoff
  - potential to clump unrelated species in same OTU

## Recommendations
  - best practices for running
    - minimum sample sizes for recovering rare species
      - we were able to correctly identify a low-abundance taxa of 5 reads to family level, compared to 50 other taxa at 2000 reads (1:400). total library size of around 100K reads. (with nanoclust min cluster size = 2)
      - (could identify taxa with 10 reads to genus level.)
      - VSEARCH had split this taxa into multiple OTUs (size ~2)
      - NanoCLUST only assigned at family level due to representative sequence, could be resolved by assigning more sequences
      - drawback
        - overestimated the number of species 100 (182 OTUs) vs 55
        - high number of unclassified OTUs
      - potential use case of identifying rare pathogens

## Future work
  - taking advantage of long amplicon (LSU, SSU)
    - aid unclassified species
    - phylogenetic methods