---
title: "Supplemental Figures"
---


```{r}
#| message: false
#| label: tbl-samplesheet
library(dplyr)

knitr::kable(
  read.csv('../../../experiments/66-fungal-isolate-ONT/sample-stats-fmt.csv') %>%
    rename_with(\(col) gsub('\\.', ' ', col)),
  align = c('r', 'l', 'r', 'r'),
  caption = "Samplesheet showing the species, sequencing barcode, number of raw reads after basecalling and number of reads remaining after quality control (QC). Corrections to names based on the latest taxonomic synonyms are shown in parenthesis."
)
```

:::{#fig-nanoclustSplitting2}
![](analysis/images/06-cluster-splitting-nanoclust-with-tax-1-30.png)
![](analysis/images/06-cluster-splitting-nanoclust-with-tax-31.png)

Chart showing how 58 fungal isolates are split into clusters based on a single execution of the even abundance mock scenario.
Each isolate has been subsampled to 2000 reads and then clustered using the NanoCLUST (UMAP + HDBSCAN) method with a minimum cluster size set to 580 (0.5% of the total library size).
Bars indicate the abundance of a cluster (number of reads). The taxonomic classification given by dnabarcoder (with the UNITE 2024 database) for each cluster is shown in the X axis labels. Green indicates that the assignment given to a cluster matches the expected species-level classification. Yellow indicates a correct genus-level classification.
Orange indicates a correct family-level classification. Red indicates the classification is incorrect for family-level and above.
Grey indicates that the cluster could not be given a taxonomic classification at all.

Clusters per isolate - Even Abundance (Scenario 1) - NanoCLUST

:::
