---
title: "Introduction"
---

```{r libraryImports}
#| message: false
library(dplyr)
library(flextable)

set_flextable_defaults(
  font.size = 10,
  theme_fun = theme_vanilla
)
```

Fungi form the third-largest kingdom of life and is estimated to
contain between two and three million species [@niskanenPushingFrontiersBiodiversity2023]. To date, approximately
161,000 species have been formally described [@indexfungorumpartnershipSpeciesFungorum2024]. Fungi have diverse living
forms, are widely
distributed and take on numerous ecological roles in terrestrial and
aquatic habitats [@peayDimensionsBiodiversityEarth2016; @Phukhamsakda2022].
Saprotrophic fungi break down
organic matter and are involved in nutrient cycling, while others form
mutually beneficial interactions with other species such as the
formation of mycorrhiza between plants and fungi [@Frey2019]. Other
fungi form pathogenic interactions causing considerable damage to
human health and agriculture [@C.2020].

When performing ecological studies, it is important to reliably detect
and identify fungi in a given environment. Metabarcoding is a technique
for identifying the organisms that are present in an environment by
sampling and analysing genomic content after amplification of
DNA barcodes [@Taberlet2018]. Applications of metabarcoding include: the
profiling of communities, analysis of diversity and inferring
ecological patterns [@Kisand2012; @Corrales2021; @Taberlet2018].

Fungi can be identified using morphological features of sporing (or
asexual) structures or by isolating them in culture from the environment
of interest [@Allen1976]. The presence of reproductive structures is
often brief and a vast number of fungal lineages (such as
ectomycorrhizal fungi) are difficult to isolate in culture. DNA
barcoding provides a solution to this problem by detecting the DNA of
targeted organisms without direct observation [@Hajibabaei2011].

Advances in high-throughput sequencing (HTS) has provided the capacity
to identify many species from bulk environmental samples such as
soil [@Tedersoo2022]. This procedure (metabarcoding) works by
extracting the genomic DNA of all organisms within a sample,
followed by the amplification of a barcode region of interest using the polymerase chain reaction
(PCR) [@Lindahl2013]. The amplified DNA is sequenced by a HTS platform such as Illumina, PacBio or Oxford
Nanopore.
Using computational techniques, sequences can be analysed and compared with reference databases to identify taxa.
The accuracy of these taxonomic classifications depends heavily on the quality of the
reference data used [@Tedersoo2022].

:::{#fig-metabarcoding-workflow}
![](resources/images/fungal-metabarcoding.png)

The main steps for a fungal metabarcoding workflow along
with a depiction of the nuclear ribosomal RNA operon. Short-read technologies target the ITS1 or ITS2 subregion
in isolation, while long-read technologies are capable of targeting the entire ITS region and beyond [@Nilsson2019a]
:::

The barcode region most commonly used for metabarcoding of fungi is the
internal transcribed spacer (ITS) region which is found between the 18S
and 28S subunits of the nuclear ribosome, also known as the small
subunit (SSU) and large subunit (LSU) respectively [@Tedersoo2022]. The
ITS region has become the universal marker for barcoding fungi for
multiple reasons. PCR amplification of the ITS region has a high success
rate due to highly conserved regions flanking the gene where PCR primers
can be designed to cover a high range of Fungi [@Schoch2012]. See
@fig-metabarcoding-workflow for a general overview of fungal
metabarcoding. Because they are non-coding, the ITS regions are highly
variable and provide high inter- and intraspecies
resolution [@Schoch2012]. Further discussion on the advantages and drawbacks of using the ITS
region can be found in @sec-fungal-marker-regions.

HTS technologies have been adopted for
metabarcoding due to their large sequencing capacity and their ability
to characterize DNA sequences from mixed samples in comparison to
traditional methods such as Sanger sequencing [@Nilsson2019a].
Short-read HTS methods such as Illumina MiSeq are commonly used for
metabarcoding studies and allow for the sequencing of up to 2x300
(forward and reverse direction) basepairs (bp) per
read [@Tedersoo2022; @Inc.2023]. In fungal metabarcoding, these read
length limitations makes Illumina best suited to analyse the smaller
ITS1 or ITS2 subregions separately with a very high sequencing
depth [@Nilsson2019a]. Illumina MiSeq, NextSeq and HiSeq produce a very
high numbers of reads per run (up to 25 million, 1.2 billion and 6
billion respectively) that results in high quality denoised
sequences [@Inc.2023; @IlluminaInc.2016].

Sequencing platforms such as Oxford Nanopore Technologies (ONT) and
PacBio provide much longer read lengths compared to Illumina HTS
methods. ONT sequencing is comparatively cheap and can be performed in a
basic molecular lab and even in the field [@Mafune2020; @Baloglu2021].
Until recently, the high error rate made nanopore sequencing unsuitable
for metabarcoding studies [@Zhang2023]. The prospect of long-read
sequencing for fungal metabarcoding is that the longer reads will be able
to cover multiple marker genes, including the full ribosomal operon, and
can provide more reliable phylogenetic placement and discovery of new
fungal lineages [@Tedersoo2018].
Sequencing technologies will be discussed further in @sec-sequencing-methods.

In the existing research, studies have explored the use of ONT
sequencing for metabarcoding of soil microbial community
structure [@Wiryawan2022], rapid pathogen
detection [@Hong2020; @Ohta2023], and testing the use
of ONT for metabarcoding of mock fungal
communities [@Baloglu2021; @Mafune2020; @langsiriTargetedSequencingAnalysis2023].
In this study, we integrated common approaches to fungal metabarcoding into a bioinformatic pipeline with a focus on analysing ONT data.
We validated our metabarcoding approaches with mock fungal communities and demonstrated how the pipeline performed on real environmental soil samples.
